lab_5_scraper/scraper.py

```diff
@@ -129,11 +252,40 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        for title_tag in article_bs.find_all("h3", class_="news-title"):
+            link_tag = title_tag.find("a", href=True)
+            if not link_tag:
+                continue
+
+            href = link_tag["href"]
+            if not href.startswith("/news"):
+                continue
+
+            full_url = "https://www.baikal-daily.ru" + str(href)
+            if full_url not in self.urls:
+                return full_url
+
+        return ""
 
     def find_articles(self) -> None:
         """
         Find articles.
         """
+        for seed_url in self.config.get_seed_urls():
+            try:
+                response = make_request(url=seed_url, config=self.config)
+                response.raise_for_status()
+            except requests.HTTPError:
+                continue
+
+            page_bs = BeautifulSoup(response.text, 'lxml')
+
+            extracted_url = self._extract_url(page_bs)
```

Comment:
lets change the logic. You should scrap divs in this function, then iterate over them and give to `self._extract_url(...)`